crates/kitsune-core/src/activitypub/fetcher.rs

```diff
@@ -88,6 +92,27 @@ pub struct Fetcher {
 }
 
 impl Fetcher {
+    async fn fetch_ap_resource<T>(&self, url: &str) -> Result<T>
+    where
+        T: DeserializeOwned + RdfNode,
+    {
+        let response = self.client.get(url).await?;
+        let content_type = response
+            .headers()
+            .get(CONTENT_TYPE)
+            .map(HeaderValue::to_str)
+            .transpose()?;
```

Comment:
Well, the equality check should already fail at the basic comparison of the union discriminator, so in the worst case, right now it's two `usize` comparison vs. one `usize` comparison with an early return.

<sub>If we want to get really technical, we could say that the bad path of content type available but wrong value then actually needs an additional `usize` compare, making it three `usize` comparisons and two partial fast-fail byte comparisons in total heh</sub>

Reply:
Ah, I meant it as a refactoring rather than a micro-optimisation. With `content_type` being an `Option`, if you are going to make the content sniffing more complicated, you would end up writing something like `content_type.map_or(false, |ct| ct.starts_with("application/ld+json") /* && … */)`, which isn't quite readable.

Reply:
Implemented this using the `headers` crate and an let-else statement